pynamodb/attributes.py

```diff
@@ -334,6 +357,52 @@ def __ne__(self, other: Any) -> bool:
         return self is not other
 
 
+class DiscriminatorAttribute(Attribute[type]):
+    attr_type = 'S'
+
+    def __init__(self, attr_name: Optional[str] = None):
+        super().__init__(attr_name=attr_name)
+        self.class_map: Dict[type, Any] = {}
```

Comment:
why `Any` and not `str`?

Reply:
If possible I'd like to support subclassing so that users can serialize other types (e.g. lists) with minimal effort.